File: go-solutions/1657_close_strings_test.go
package leetcode

import (
	"fmt"
	"testing"
)

type closeStringsTest struct {
	word1 string
	word2 string
	want  bool
}

var closeStringsTests = []closeStringsTest{
	{"abc", "bca", true},
	{"a", "aa", false},
	{"cabbba", "abbccc", true},
	{"abbzzca", "babzzcz", false},
	{"uau", "ssx", false},
}

func TestCloseStrings(t *testing.T) {
	for _, c := range closeStringsTests {
		t.Run(fmt.Sprintf("%v_%v_%v", c.word1, c.word2, c.want), func(t *testing.T) {
			if got := CloseStrings(c.word1, c.word2); got != c.want {
				t.Errorf("got %t, want %t", got, c.want)
			}
		})
	}
}

File: ts-solutions/tests/1657-close-strings.test.ts
import { closeStrings } from '../src/1657-close-strings'

type CloseStringsTest = {
	word1: string
	word2: string
	want: boolean
}

const closeStringsTest: CloseStringsTest[] = [
	{ word1: `abc`, word2: `bca`, want: true },
	{ word1: `a`, word2: `aa`, want: false },
	{ word1: `cabbba`, word2: `abbccc`, want: true },
]

describe(`closeStrings`, () => {
	test.each(closeStringsTest)(
		`closeStrings($word1, $word2) => $want`,
		({ word1, word2, want }) => {
			expect(closeStrings(word1, word2)).toBe(want)
		}
	)
})
